File: src/hash.ts
import { createHash } from "crypto";

export function hashString(input: string): string {
  return hashBytes(Buffer.from(input));
}

export function hashBytes(input: Buffer | Uint8Array): string {
  return createHash("sha256").update(input).digest("hex");
}

export function hashStrings(parts: string[]): string {
  return hashString(parts.join("\n\n"));
}

File: internal/context/hash.go
package context

import (
	"crypto/sha256"
	"encoding/hex"
)

func HashString(input string) string {
	return HashBytes([]byte(input))
}

func HashBytes(input []byte) string {
	hash := sha256.Sum256(input)
	return hex.EncodeToString(hash[:])
}

func HashStrings(parts []string) string {
	combined := ""
	for i, part := range parts {
		if i > 0 {
			combined += "\n\n"
		}
		combined += part
	}
	return HashString(combined)
}
